Userland/Libraries/LibCompress/Huffman.h

```diff
@@ -12,35 +12,34 @@
 
 namespace Compress {
 
-template<size_t Size>
-void generate_huffman_lengths(Array<u8, Size>& lengths, Array<u16, Size> const& frequencies, size_t max_bit_length, u16 shift = 0)
+inline void generate_huffman_lengths(Bytes lengths, ReadonlySpan<u16> frequencies, size_t max_bit_length, u16 shift = 0)
 {
-    VERIFY((1u << max_bit_length) >= Size);
-    u16 heap_keys[Size]; // Used for O(n) heap construction
-    u16 heap_values[Size];
+    VERIFY(lengths.size() == frequencies.size());
+    auto const size = lengths.size();
+    VERIFY((1u << max_bit_length) >= size);
+    u16 heap_keys[size]; // Used for O(n) heap construction
+    u16 heap_values[size];
 
-    u16 huffman_links[Size * 2] = { 0 };
+    u16 huffman_links[size * 2];
```

Comment:
If I see that correctly that's now a VLA,
didnt we forbid those?

Reply:
I don't know!

Does a VLA have very different semantics than a statically sized array where the size depends on a template parameter? Most of https://nullprogram.com/blog/2019/10/27/ probably applies to the lhs too?

Reply:
> I don't know!

(...but afaik there's -Wvla, which we don't seem to use)

Reply:
I think that we discourage the usage of VLA. What about using a Vector with an inline capacity?
I'll do a full review later on.

Reply:
`rg -i vla Documentation` doesn't find anything at least.

IMHO this PR doesn't really change the behavior of this function WRT VLAs since it's a template-size-dependent array on the RHS too.

(But if there's some way in which this is worse than that, do tell! I suppose in the LHS it's very clear that the array size is not attacker-controllable since all possible sizes are known at compile time. But we still use a bounded size for these arrays.)

Reply:
I read a little bit further and indeed, I can confirm that it is currently perfectly safe.
While a vector with inline capacity would be a little bit more future-proof (if users start to use the function with dynamic containers), it's not a huge deal either so do whatever you want.

Reply:
Resolving this then since it's unrelated to this PR :)